adapter/service/virtualmeter/manager.go

```diff
@@ -64,18 +56,18 @@ func (m *managerWrapper) WithAdapter(ad adapter.Adapter) {
 
 // RegisterThing creates a virtual meter and numeric meter services for a thing based on the existing
 // services. VMS is then added to a think and numeric is added based on whether the virtual meter is already active.
-func (m *managerWrapper) RegisterThing(thing adapter.Thing, publisher adapter.Publisher) error {
+func (m *manager) RegisterThing(thing adapter.Thing, publisher adapter.Publisher) error {
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
-	if len(thing.InclusionReport().Groups) == 0 {
-		vmsSpec, numericSpec := m.createVirtualServicesForThing(thing, "")
+	for _, group := range thing.InclusionReport().Groups {
+		vmsSpec, numericSpec := m.createVirtualServicesForThing(thing, group)
 
-		return m.registerVirtualServices(thing, publisher, vmsSpec, numericSpec)
-	}
+		if vmsSpec == nil || numericSpec == nil {
+			continue
+		}
 
-	for _, group := range thing.InclusionReport().Groups {
-		vmsSpec, numericSpec := m.createVirtualServicesForThing(thing, "_"+group)
+		log.Infof("registering virtual service")
```

Comment:
Either remove this log or follow convention of "where: what". Also if you log this, add some details, like thing address and maybe group for which it is registered.